src/ExpressionBuilder.ts

```diff
@@ -10,6 +10,7 @@ import type { DataNode } from './VariablePolyfill';
 import { splitExpression } from './ExpressionSplitter';
 import type { ExpressionCode, ExpressionText } from './ExpressionSplitter';
 import { parseWithEsprimaNext } from './Parser';
+import type { TournamentHooks } from './';
```

Comment:
since this is only a `type`, it should be okay, but as a general practice, we should avoid importing stuff from barrel files.

Reply:
I'd forgotten I'd done that! I was just trying to get it working, so I meant to come back to it.